validation-utils/src/main/java/org/hypertrace/config/validation/IpValidationUtils.java

```diff
@@ -0,0 +1,43 @@
+package org.hypertrace.config.validation;
+
+import static java.math.BigInteger.ONE;
+
+import inet.ipaddr.IPAddress;
+import inet.ipaddr.IPAddressString;
+import inet.ipaddr.IPAddressStringParameters;
+
+public class IpValidationUtils {
+  private static final IPAddressStringParameters ADDRESS_VALIDATION_PARAMS =
+      new IPAddressStringParameters.Builder()
+          // Allows ipv4 joined segments like 1.2.3, 1.2, or just 1 For the case of just 1 segment
+          .allow_inet_aton(false)
+          // Allows an address to be specified as a single value, eg ffffffff, without the standard
+          // use of segments like 1.2.3.4 or 1:2:4:3:5:6:7:8
+          .allowSingleSegment(false)
+          // Allows zero-length IPAddressStrings like ""
+          .allowEmpty(false)
+          .toParams();
+
+  public static boolean isValidSubnet(String subnet) {
+    IPAddress parsedAddress = parseAddress(subnet);
+    if (parsedAddress == null) {
+      return false;
+    }
+    return !parsedAddress
```

Comment:
So this is saying if in CIDR notation, we don't allow representing a single IP? If so, why not? Seems like these prefix checks are superfluous, but I'll admit I'm not exactly following

Reply:
> So this is saying if in CIDR notation, we don't allow representing a single IP? If so, why not?

Yes, it is a valid IP address but not a valid subnet. That's how the implementation is in the existing ip-utils.

Reply:
The great thing about not having a shared lib here is we can tailor this to what's relevant here. When the use case allows either, it's silly to parse it twice to figure out if it's valid. It's more compute and it's harder to read the validator code both here and where it's called.

So this whole file could really just be
```java
boolean isValidIpAddressOrSubnet(String input) {
  return new IPAddressString(address, ADDRESS_VALIDATION_PARAMS).getAddress() != null;
}
```